tests/unit/challenges/test_views.py

```diff
@@ -3005,3 +3005,52 @@ def test_star_challenge(self):
         response = self.client.post(self.url, {})
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class GetChallengePhaseByPk(BaseChallengePhaseClass):
+
+    def setUp(self):
+        super(GetChallengePhaseByPk, self).setUp()
+        self.url = reverse_lazy('challenges:get_challenge_phase_by_pk',
+                                kwargs={'pk': self.challenge_phase.pk})
+
+    def test_get_challenge_phase_by_pk(self):
+        expected = {
+            "id": self.challenge_phase.id,
+            "name": self.challenge_phase.name,
+            "description": self.challenge_phase.description,
+            "leaderboard_public": self.challenge_phase.leaderboard_public,
+            "start_date": "{0}{1}".format(self.challenge_phase.start_date.isoformat(), 'Z').replace("+00:00", ""),
+            "end_date": "{0}{1}".format(self.challenge_phase.end_date.isoformat(), 'Z').replace("+00:00", ""),
+            "challenge": self.challenge_phase.challenge.pk,
+            "is_public": self.challenge_phase.is_public,
+            "is_active": True,
```

Comment:
@Sanji515 can't we fetch this from `self.challenge_phase` instead of hardcoding it?

Reply:
@Ram81 Actually the `response.data` is a dictionary and `self.challenge_phase` is a `queryset` so for comparing we need both of them to be a dictionary.

Reply:
@Ram81 We can't fetch `is_active` from `self.challenge_phase` because it`(is_active)` came from `serializers.py` not from `models.py`.